disassemblers/ofrak_pyghidra/src/ofrak_pyghidra/components/pyghidra_components.py

```diff
@@ -91,39 +91,39 @@ def __init__(
         self.analysis_store = analysis_store
 
     async def analyze(self, resource: Resource, config: PyGhidraAutoAnalyzerConfig = None):
-        with TemporaryDirectory() as tempdir:
-            program_file = os.path.join(tempdir, "program")
-            await resource.flush_data_to_disk(program_file)
-            if config is None:
-                decomp = False
-                language = None
-            else:
-                decomp = config.decomp
-                language = config.language
-            for tag in _GHIDRA_AUTO_LOADABLE_FORMATS:
-                if resource.has_tag(tag):
-                    self.analysis_store.store_analysis(
-                        resource.get_id(), unpack(program_file, decomp, language)
-                    )
-                    return PyGhidraProject()
-
-            program_attrs = resource.get_attributes(ProgramAttributes)
-            # Guess that the base address is the min start address of any memory region
-            regions = await resource.get_children_as_view(
-                MemoryRegion, r_filter=ResourceFilter.with_tags(MemoryRegion)
-            )
-            base_address = min(code_region.virtual_address for code_region in regions)
-
-            self.analysis_store.store_analysis(
-                resource.get_id(),
-                unpack(
-                    program_file,
-                    decomp,
-                    language=_arch_info_to_processor_id(program_attrs),
-                    base_address=base_address,
-                ),
-            )
-            return PyGhidraProject()
+        tempdir = mkdtemp(prefix="rbs-pyghidra-bin")
```

Comment:
These calls to `mkdtemp` will leak tempfiles, right? 

If so, probably not an issue if we can't work around it while still using the caching results across runs. But we should consider if there is an easy way to clean up when OFRAK terminates. Not worth blocking the PR over; just worth thinking about.